applications/ConvectionDiffusionApplication/custom_elements/qs_convection_diffusion_explicit.cpp

```diff
@@ -214,6 +214,53 @@ void QSConvectionDiffusionExplicit<3,4>::CalculateMassMatrix(
 /***********************************************************************************/
 /***********************************************************************************/
 
+template<>
+void QSConvectionDiffusionExplicit<2,3>::CalculateLumpedMassVector(
+    VectorType& rLumpedMassVector,
+    const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_TRY;
+
+    // Define local variables
+    const unsigned int local_size = 3;
+    const double one_third_area = GetGeometry().Area() / 3.0;
+    // Initialize and calculate elemental lumped mass vector
+    if (rLumpedMassVector.size() != local_size) {
+        rLumpedMassVector.resize(local_size, false);
+    }
+    for (IndexType i = 0; i < local_size; ++i) {
+        rLumpedMassVector(i) = one_third_area;
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+
+template<>
+void QSConvectionDiffusionExplicit<3,4>::CalculateLumpedMassVector(
+    VectorType& rLumpedMassVector,
+    const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_TRY;
+
+    // Define local variables
+    const unsigned int local_size = 4;
+    const double one_fourth_volume = GetGeometry().Volume() / 4.0;
+    // Initialize and calculate elemental lumped mass vector
+    if (rLumpedMassVector.size() != local_size) {
+        rLumpedMassVector.resize(local_size, false);
+    }
+    for (IndexType i = 0; i < local_size; ++i) {
+        rLumpedMassVector(i) = one_fourth_volume;
+    }
+
+    KRATOS_CATCH("");
+}
```

Comment:
You don't need the template specialization in this case. Check the implementation in the compressible element.

Reply:
Thanks. Your implementation is way more clever than mine.. I will push it now.